aioxmpp/avatar/service.py

```diff
@@ -748,33 +807,107 @@ def publish_avatar_set(self, avatar_set):
                     avatar_set.metadata,
                     id_=id_
                 )
-            finally:
-                if self._synchronize_vcard:
-                    my_vcard = yield from self._vcard.get_vcard()
-                    my_vcard.set_photo_data("image/png",
-                                            avatar_set.image_bytes)
-                    self._vcard_id = avatar_set.png_id
-                    yield from self._vcard.set_vcard(my_vcard)
-                    yield from self._presence_server.resend_presence()
+                done = True
+
+            if self._synchronize_vcard:
+                my_vcard = yield from self._vcard.get_vcard()
+                my_vcard.set_photo_data("image/png",
+                                        avatar_set.image_bytes)
+                self._vcard_id = avatar_set.png_id
+                yield from self._vcard.set_vcard(my_vcard)
+                yield from self._presence_server.resend_presence()
```

Comment:
``resend_presence`` is not a coroutine. It can return a (awaitable) stanza token, but it may also return None.

I also don’t think that waiting for the resend is necessary.

Reply:
oops, should have read the docs better, good catch!